### PR TITLE
dev-python/influxdb: add Python 3.7

### PR DESCRIPTION
**Pull request by Hawk777**

### CODE REVIEW

**Comment by Hawk777** [comment]
A comment on <https://github.com/influxdata/influxdb-python/issues/794> claims that this package has been tested on Python 3.7. I also did some basic things myself which worked. I couldn’t get the full test suite passing, but have filed <https://github.com/influxdata/influxdb-python/issues/799> and <https://github.com/influxdata/influxdb-python/issues/798> related to issues I ran into trying to do so.

File: dev-python/influxdb/influxdb-5.2.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Copyright 1999-2020 Gentoo Authors
```
Comment (mgorny) [inline dev-python/influxdb/influxdb-5.2.3.ebuild:1]: Please look up on how to write good commit messages.

File: dev-python/influxdb/influxdb-5.2.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	amd64? ( test? (
+		dev-db/influxdb
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/nose[${PYTHON_USEDEP}]
+		dev-python/pandas[${PYTHON_USEDEP}]
+		dev-python/requests-mock[${PYTHON_USEDEP}]
+	) )"
+
+python_test() {
```
Comment (mgorny) [inline dev-python/influxdb/influxdb-5.2.3.ebuild:37]: Use `distutils_enable_tests`.

File: dev-python/influxdb/influxdb-5.2.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	!test? ( test )"
+
+RDEPEND="
+	>=dev-python/python-dateutil-2.6.0[${PYTHON_USEDEP}]
+	>=dev-python/six-1.10.0[${PYTHON_USEDEP}]
+	dev-python/pytz[${PYTHON_USEDEP}]
+	>=dev-python/requests-2.17.0[${PYTHON_USEDEP}]
+"
+DEPEND="${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/influxdb/influxdb-5.2.3.ebuild:28]: Remove the redundant dep.

File: dev-python/influxdb/influxdb-5.2.3.ebuild (hunk 1 above)
Comment (Hawk777) [reply dev-python/influxdb/influxdb-5.2.3.ebuild:1]: I was under the impression that version bumps in the Gentoo repo typically didn’t say anything about what changed in the underlying package, and that you had to go upstream to find that out. But sure, I’ve added a little bit more text in beea73e70534; let me know if there is anything further you feel is worth putting in the message.

File: dev-python/influxdb/influxdb-5.2.3.ebuild (hunk 2 above)
Comment (Hawk777) [reply dev-python/influxdb/influxdb-5.2.3.ebuild]: beea73e70534

File: dev-python/influxdb/influxdb-5.2.3.ebuild (hunk 3 above)
Comment (Hawk777) [reply dev-python/influxdb/influxdb-5.2.3.ebuild]: beea73e70534

File: dev-python/influxdb/influxdb-5.2.3.ebuild (hunk 1 above)
Comment (mgorny) [reply dev-python/influxdb/influxdb-5.2.3.ebuild:1]: But they usually do say what the new version is.

**Comment by Hawk777** [comment]
Nope, and I [reported it upstream](https://github.com/influxdata/influxdb-python/issues/798).

**Comment by Hawk777** [comment]
Hi, just checking if anything further is needed? AFAICT I responded to all questions/comments. Thanks!

**Comment by juippis** [comment]
Well this package is gone after 7622f900c65

It needs to be re-added preferably after upstream has fixed reported issues, or put to ::guru for now.
(I guess?)

**Comment by Hawk777** [comment]
So you object to the addition of a version bump with failing tests? I have to wait until another release, hopefully with the tests fixed?

**Comment by juippis** [comment]
Truthfully I was just doing some cleaning in Github and noticed this package has been removed, checked in, saw some errors and link to upstream. I have no idea why python team hasn't merged this and didn't dig deeper into it either. 

In **my** eyes adding it now counts as a "new package" and new package additions should be flawless. But python team can decide otherwise.

**Comment by Hawk777** [comment]
> should be flawless

It seems that “package has failing tests” came up on `gentoo-dev` recently, too. I can always just change the ebuild to `RESTRICT=test` if you want? Or add a pile of patches, I guess, though I do prefer sticking close to upstream.

**Comment by Hawk777** [comment]
I’ve rebased this. Still awaiting comment on if any further work is needed, or what that work should be.

**Comment by Hawk777** [comment]
In the rebase I also changed myself to maintainer, since apparently this PR has transformed itself from a version bump into a new package.

**Comment by Hawk777** [comment]
@mgorny @juippis I rebased this and changed it to version 5.3.0 (which upstream released since I started this PR initially). That upstream version change added `dev-python/msgpack` as a dependency (which I added to the ebuild, of course) and fixed the test suite problem I indicated above. However the test suite is now broken for [a different reason](https://github.com/influxdata/influxdb-python/issues/714), but that was easy to work around. The test suite passes for me.

I have not added 3.8. I really tried to, but in order to add 3.8 to this package I would have to add 3.8 to `dev-python/pandas` (because the test suite uses it), in order to add 3.8 to `dev-python/pandas` I would have to add 3.8 to `dev-python/openpyxl`, and GH-14481 already did that in January but has not been merged yet, and it in turn depends on GH-14349 which was also filed in January and not merged yet. I didn’t explore the rest of the deptree of Pandas to see if there were any other packages that would also need bumping.

**Pull request closed without merging**